### PR TITLE
speakeasy: Update definition for missing functions

### LINKED ISSUE #21255: DefinitelyTyped/types/speakeasy/ typing missing for some functions
If you know how to fix the issue, make a pull request instead.

- [ ] I tried using the `@types/speakeasy` package and had problems.

- Authors: Lucas Woo <https://github.com/legendecas>, Alexander Batukhtin <https://github.com/mrOlorin> 

The typing works only for the 

```
export declare function digest(options: DigestOptions): string;

export declare function generate_key(options: GenerateOptions): Key;

export declare function generateSecret(options: GenerateSecretOptions): Key;

export declare function generateSecretASCII(length?: number, symbols?: boolean): string;

export declare function otpauthURL(options: OtpauthURLOptions): string;

export declare function hotp(options: HotpOptions): string;

export declare function counter(options: HotpOptions): string;

export declare function totp(options: TotpOptions): string;

export declare function time(options: TotpOptions): string;
```

but there are some methods with the totp , hotp and so on. 

```
import { totp, generateSecret, TotpOptions } from "speakeasy";
export class TOTPHelper {
    static generate(): string {
        var secret = generateSecret({
            length: 20
        });

        return secret.base32;
    }
    static validate(token: string, key: string) {

        var verified = totp.verify({
            secret: key,
            encoding: 'base32',
            token: token
        });

        return verified;
    }
}
```

in this, `generateSecret` have typings but  `totp.verify` is not.

### PR DESCRIPTION
**Pull request by xeoneux**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/speakeasyjs/speakeasy/issues/90
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

  Fixes #21255

### CODE REVIEW

File: types/speakeasy/index.d.ts [hunk 1]
```diff
@@ -1,9 +1,17 @@
 // Type definitions for speakeasy v2.0.0
-// Project: https://github.com/markbao/speakeasy
-// Definitions by: Lucas Woo <https://github.com/legendecas>, Alexander Batukhtin <https://github.com/mrOlorin>
+// Project: https://github.com/speakeasyjs/speakeasy
+// Definitions by: Lucas Woo <https://github.com/legendecas>, Alexander Batukhtin <https://github.com/mrOlorin>, Aayush Kapoor <https://github.com/xeoneux>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-interface Key {
+declare module "speakeasy" {
```
Comment (mhegazy) [inline types/speakeasy/index.d.ts]: Do not use `declare module "speakeasy"` instead use `export as namespace speakeasy`

**Review by mhegazy — CHANGES_REQUESTED** [review]

**Comment by xeoneux** [comment]
@mhegazy how do I handle a function that is also an object with method?

```
function totp(options: TotpOptions): string;
function totp.verify(options: TotpVerifyOptions): boolean;
function totp.verifyDelta(options: TotpVerifyOptions): boolean;
```

**Comment by xeoneux** [comment]
@mhegazy solved! 😄

**Review by mrOlorin — APPROVED** [review]

File: types/speakeasy/index.d.ts [hunk 2]
```diff
@@ -1,94 +1,121 @@
...
-interface Key {
-    ascii: string;
-    base32: string;
-    hex: string;
-    qr_code_ascii: string;
-    qr_code_hex: string;
-    qr_code_base32: string;
-    google_auth_qr: string;
-    otpauth_url?: string;
+export as namespace speakeasy;
```
Comment (RyanCavanaugh) [inline types/speakeasy/index.d.ts]: This isn't a UMD module, so there shouldn't be a UMD module declaration here

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

**Comment by xeoneux** [comment]
@RyanCavanaugh fixed! 😸

**Pull request merged**